cpp/src/arrow/type_traits.h

```diff
@@ -998,6 +998,21 @@ constexpr bool is_run_end_type(Type::type type_id) {
   return false;
 }
 
+/// \brief Check for a time type
+///
+/// \param[in] type_id the type-id to check
+/// \return whether type-id is a primitive type one
+constexpr bool is_time(Type::type type_id) {
```

Comment:
1) can you move this somewhere near `is_temporal`?
2) should we also define `is_date`?

Reply:
1. Yes
2. Yes